src/category_theory/limits/shapes/zero.lean

```diff
@@ -60,22 +61,22 @@ class has_zero_object :=
 (unique_to : Π X : C, unique (zero ⟶ X))
 (unique_from : Π X : C, unique (X ⟶ zero))
 
-variables [has_zero_object.{v} C]
-
 variables {C}
 
+namespace has_zero_object
+
+variables [has_zero_object.{v} C]
+
 /--
 Construct a `has_zero C` for a category with a zero object.
 This can not be a global instance as it will trigger for every `has_zero C` typeclass search.
 -/
-def zero_of_zero_object : has_zero C :=
+def has_zero : has_zero C :=
```

Comment:
`protected`?

Reply:
Okay. I guess I don't quite understand the criteria for marking something `protected`. I understand that it means that the name isn't visible just by opening the namespace.